### PR TITLE
OCPCLOUD-2573: Enables kubelet cloud credential providers

### PR DESCRIPTION
**Pull request by theobarberbany**
Enables the DisableKubeletCloudCredentialProviders feature gate 

Cloud credential providers for AWS, GCP and Azure are now functional. We can disable the in tree kubelet providers.

### CODE REVIEW

File: config/v1/types_feature.go [hunk 1]
```diff
@@ -217,9 +214,7 @@ var defaultFeatures = &FeatureGateEnabledDisabled{
 		vSphereControlPlaneMachineset,
 		sdnLiveMigration,
 	},
-	Disabled: []FeatureGateDescription{
-		disableKubeletCloudCredentialProviders, // We do not currently ship the correct config to use the external credentials provider.
```
Comment (JoelSpeed) [inline config/v1/types_feature.go:221]: Needs to be moved into the Enabled list to be explicitly enabled

Comment (theobarberbany) [reply config/v1/types_feature.go:221]: So should it not be renamed `enableKubeletCloudCredentialProviders` to make more sense?

I'm also not following how we go from these changes to actually enabling / disabling functionality..

**Comment by theobarberbany** [comment]
GCP e2es are a bit sad at the moment, but it's being worked on :) 

I don't think the failures are related to this PR! :D

**Comment by JoelSpeed** [comment]
Changes LGTM but I need to see a pass on GCP before we can merge this

**Comment by theobarberbany** [comment]
GCP is failing due to a configuration error.

https://github.com/openshift/machine-config-operator/pull/4260 fixes that.

File: payload-manifests/featuregates/featureGate-Default.yaml [hunk 2]
```diff
@@ -0,0 +1,176 @@
+{
```
Comment (theobarberbany) [inline payload-manifests/featuregates/featureGate-Default.yaml:1]: Hmm - this has been removed.

File: payload-manifests/featuregates/featureGate-TechPreviewNoUpgrade.yaml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        "name": "cluster"
+    },
+    "spec": {
+        "featureSet": "TechPreviewNoUpgrade"
+    },
+    "status": {
+        "featureGates": [
+            {
+                "disabled": [
+                    {
```
Comment (theobarberbany) [inline payload-manifests/featuregates/featureGate-TechPreviewNoUpgrade.yaml:15]: Hmm - this has been removed.

**Comment by JoelSpeed** [comment]
/retitle OCPCLOUD-2573: Enables kubelet cloud credential providers

**Comment by theobarberbany** [comment]
So it looks the GCR image pull still fails :/ 
```
time="2024-03-20T12:15:42Z" level=info msg="resulting interval message" message="{Failed  Failed to pull image \"gcr.io/authenticated-image-pulling/alpine:3.7\": reading manifest 3.7 in gcr.io/authenticated-image-pulling/alpine: denied: Permission denied for \"3.7\" from request \"/v2/authenticated-image-pulling/alpine/manifests/3.7\".  map[reason:Failed]}"
time="2024-03-20T12:15:42Z" level=info msg="resulting interval message" message="{BackOff  Back-off pulling image \"gcr.io/authenticated-image-pulling/alpine:3.7\" map[interesting:true reason:BackOff]}"
```

**Comment by theobarberbany** [comment]
Interestingly, the previously failing tests from the previous CI run: 

```
: [sig-apps] ReplicaSet should serve a basic image on each replica with a private image [Suite:openshift/conformance/parallel] [Suite:k8s]
: [sig-apps] ReplicaSet should serve a basic image on each replica with a public image [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
```

Are now running and passing. 

The Image failing to pull `gcr.io/authenticated-image-pulling/alpine:3.7` should be failing: https://github.com/openshift/origin/blob/2cecd5f51c4f4b51b255bcadd294d208cbd00a9b/pkg/monitortests/node/legacynodemonitortests/kubelet.go#L527-L528

**Comment by theobarberbany** [comment]
Both QE and I have tested this on GCP: 

Manually exec'ing the binary: 
```
[root@tbarberb-dev-4thxv-worker-a-59jj5 ~]# cat test.json
{
  "apiVersion": "kubelet.k8s.io/v1",
  "kind": "CredentialProviderRequest",
  "image": "gcr.io/openshift-gce-devel/hello-ecr"
}

[root@tbarberb-dev-4thxv-worker-a-59jj5 ~]# cat test.json | /usr/libexec/kubelet-image-credential-provider-plugins/gcr-credential-provider get-credentials --v 3
I0322 12:34:29.860294   18330 getcredentials.go:110] get-credentials (authFlow gcr)
{"kind":"CredentialProviderResponse","apiVersion":"credentialprovider.kubelet.k8s.io/v1","cacheKeyType":"Image","cacheDuration":"0s","auth":{"*.gcr.io":{"username":"_token","password":"..."},"*.pkg.dev":{"username":"_token","password":"..."},"container.cloud.google.com":{"username":"_token","password":"..."},"gcr.io":{"username":"_token","password":"..."}}}
```
After manually enabling the featureagate, and waiting for the MachineConfigPools to roll the image pull succeeds (     imagePullPolicy: Always)
```
~/projects/4.16 on ☁️   tbarberb@redhat.com
❯ oc new-app --allow-missing-images --image gcr.io/openshift-gce-devel/hello-ecr:latest
~/projects/4.16 on ☁️   tbarberb@redhat.com took 5s
❯ oc edit deployment hello-ecr
deployment.apps/hello-ecr edited

~/projects/4.16 on ☁️   tbarberb@redhat.com took 4s
❯ oc get deployment
NAME        READY   UP-TO-DATE   AVAILABLE   AGE
hello-ecr   3/3     3            3           70m
```

**Comment by sunzhaohua2** [comment]
Tested on gcp cluster and azure cluster, pull image from GCR and ACR succeed, run critical cases passed.
/label qe-approved

**Comment by JoelSpeed** [comment]
/test e2e-gcp

Cluster failed to bootstrap

**Comment by JoelSpeed** [comment]
Private registry tests appear to be passing, among other inconsistent and spurious failures
/lgtm

**Review by damdo — COMMENTED** [review]
/lgtm

**Pull request merged**